### ISSUE TITLE
non-passive event listeners

### ISSUE THREAD

**Issue by jwaliszko** [openseadragon/openseadragon#1669]
New Chrome prints the following warnings in verbose mode (which is related to this new feature: https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md):

_[Violation] Added non-passive event listener to a scroll-blocking 'wheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952_

Is there any chance to expose this configuration option to be setup by developers, or to hardcode it according to new standards?

**Comment by alvaromartmart** [comment]
I think it is hardcoded

https://github.com/openseadragon/openseadragon/blob/eafdc452c9e312f1f69a44bbb0db0023826da92d/src/mousetracker.js#L1343-L1348

I tried replaced the hardcoded `false` with `{passive: true}` and the warning disappeared

**Comment by msalsbery** [comment]
Coincidentally, I'm working on a MouseTracker update right now and that will be addressed!

**Comment by iangilman** [comment]
@msalsbery Wonderful!

I imagine we want to use `{ passive: false }`, since we can indeed cancel the default scrolling behavior, right?

**Comment by msalsbery** [comment]
> I imagine we want to use { passive: false }, since we can indeed cancel the default scrolling behavior, right?

@iangilman Yes sir! :)

**Comment by msalsbery** [comment]
...and word on the street is Apple is FINALLY working on implementing the W3C pointer events model, which is great news after all these years!

**Comment by iangilman** [comment]
Sweet! I'm excited!

**Comment by NEXCO-WestUSA** [comment]
> Coincidentally, I'm working on a MouseTracker update right now and that will be addressed!
> 
> Thank you for the heads-up!

When will this enhancement be released?

**Comment by msalsbery** [comment]
Hi @NEXCO-WestUSA 

Finally getting real close, and just remembered this issue when I turned on verbose console mode on the new Chromium-based Edge (love that browser!)

Will update very soon...We will need to set passive false so I’m not sure the warnings will go away, but we’ll  see...

**Comment by msalsbery** [comment]
@iangilman Thinking out loud, I'm not sure how to address that. One fix is to mark it passive to eliminate the warning, but that affects the out-of-the-box behavior of OpenSeadragon viewers, and could mean a (possibly breaking) API change...

I don't know....I know it's not crucial -the warning was added so wheel event handlers on window, document, and document.body wouldn't have their handling burdened, none of which applies to us...

**Comment by iangilman** [comment]
Yeah, we want to keep the functionality. There's no way to tell the browser, "No we really mean it, don't worry about it"?

I suppose we could add a flag to turn it off for people who care more about the warnings than the functionality, but I'm concerned that's just adding more noise to the API. I kind of think maybe if you ask for verbose warnings this is part of what you get? Anyway, I don't feel strongly either way, so long as we keep the functionality.

**Comment by msalsbery** [comment]
@iangilman  Agreed. For sure the functionality isn’t going to change, it’s just a strange warning for the browsers to add. It’s strictly informational, and if one sets the handler as passive and still calls preventDefault then the result is just another warning. So either way, our gesture handling is going to generate a warning.

I’m not real concerned about it, but I do intend to dig for discussions by the implementors on the rationale and at least update our documentation accordingly

**Comment by msalsbery** [comment]
@iangilman ...so now that I think about it, I misspoke when I stated it would change the functionality...only the warning would change.

Anyway, mostly just tracking my thoughts here for reference when I get to that part of the patch I’m currently working on...there’s already too much machinery to try to keep straight in my head  ;-)

**Comment by msalsbery** [comment]
#1872 has now landed and adds support for passive event handlers.

I'm keeping this open until I've tested some scenarios a little further...

**Comment by msalsbery** [comment]
Our wheel event handling is correct in its need to call preventDefault so can't be passive. Not anything we can do about the warning.

OpenSeadragon.addEvent() now supports the EventTarget.addEventListener() options parameter on user agents that support it. We don't use it internally as we have no hard-coded passive event handling.

Closing this...feel free to re-open if it's still an issue!

**Comment by dmolesUC** [comment]
Re @iangilman's [comment](https://github.com/openseadragon/openseadragon/issues/1669#issuecomment-665144480) about telling the browser "we meant to do that" — I've only played around with this a little bit, but it seems like if you explicitly pass `passive: false` this silences the warning, e.g.

```js
canvas.addEventListener('wheel', listener, { passive: false, capture: true })
```

**Comment by iangilman** [comment]
@dmolesUC great! Seems worth doing :) 

@msalsbery sound good? Do you want to take care of this?

**Comment by msalsbery** [comment]
@iangilman @dmolesUC 

Yes seems like it's worth doing. I'll handle it

**Comment by msalsbery** [comment]
Fixed by #2043

@dmolesUC Thank you for the great suggestion!
